### PR TITLE
chore: switch from Update() to Patch()

### LINKED ISSUE #151: Change the Update in Reconcile() to a Patch
https://github.com/kubernetes-sigs/cluster-api-provider-docker/pull/148#discussion_r310257842

`r.Client.Update(...)` is used within `controllers/dockermachine_controller.go` in `Reconcile(...)`. Change that to a `Patch`.

### PR DESCRIPTION
**Pull request by dennisme**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #151 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
- Switch from Update() to Patch()
```

### CODE REVIEW

File: controllers/dockermachine_controller.go [hunk 1]
```diff
@@ -123,7 +123,8 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-	if err := r.Client.Update(ctx, dockerMachine); err != nil {
+
+	if err := r.Client.Patch(ctx, dockerMachine, patch); err != nil {
```
Comment (detiber) [inline controllers/dockermachine_controller.go]: Odd, there is already a deferred patch happening on line 79, is this update even necessary  here?

Comment (chuckha) [inline controllers/dockermachine_controller.go]: unfortunately there is a bug in this client's patch call that requires slightly more work than this. Please see

https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/blob/master/controllers/kubeadmconfig_controller.go#L289

Comment (chuckha) [reply controllers/dockermachine_controller.go]: edit: What jason said. You can delete this line instead of modifying

Comment (dennisme) [reply controllers/dockermachine_controller.go]: Ah ok 👀

Comment (dennisme) [reply controllers/dockermachine_controller.go]: So the defer is handling the patchMachine, so the r.Client.Patch can simply be removed. That seems to follow with https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/pull/107.

File: .gitignore [hunk 2]
```diff
@@ -11,3 +11,4 @@ config/samples
 dist
 bin
 manager_image_patch.yaml-e
+cover.out
```
Comment (neolit123) [inline .gitignore]: this change seems out of scope for this PR.

Comment (dennisme) [reply .gitignore]: I can backout that change, but `make test` will produce this https://github.com/kubernetes-sigs/cluster-api-provider-docker/blob/master/Makefile#L33

Comment (chuckha) [reply .gitignore]: I'm ok with this change going through. @neolit123 do you think this is worth blocking on?
